microprofile-fault-tolerance/src/test/java/org/jboss/eap/qe/microprofile/fault/tolerance/integration/metrics/MultipleMetricsExtensionTest.java

```diff
@@ -0,0 +1,402 @@
+     *
+     * @return A ShrinkWrap {@link Archive> containing a deployment that contains MP Fault Tolerance operations
+     */
+    @Deployment(name = FAULT_TOLERANCE_DEPLOYMENT, managed = false)
+    public static Archive<?> deployment() {
+        String mpConfig = "Timeout/enabled=true";
+        return ShrinkWrap.create(WebArchive.class, FAULT_TOLERANCE_DEPLOYMENT + ".war")
+                .addPackages(true, HelloService.class.getPackage())
+                .addAsManifestResource(ConfigurationUtil.BEANS_XML_FILE_LOCATION, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    /**
+     * A deployment that enables MP Telemetry via the {@code otel.sdk.disabled=false} MP Config property.
+     *
+     * @return A ShrinkWrap {@link Archive> containing a deployment that enables MP Telemetry
+     */
+    @Deployment(name = FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY, managed = false)
+    public static Archive<?> deploymentWithMPTelemetryEnabled() {
+        String mpConfig = "otel.service.name=" + FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY
+                + "\notel.sdk.disabled=false\nTimeout/enabled=true";
+        return ShrinkWrap.create(WebArchive.class, FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY + ".war")
+                .addPackages(true, HelloService.class.getPackage())
+                .addAsManifestResource(ConfigurationUtil.BEANS_XML_FILE_LOCATION, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    /**
+     * A deployment that enables MP Telemetry via the {@code otel.sdk.disabled=false} MP Config property, and
+     * disabling MP Fault Tolerance metrics collection using Micrometer via the
+     * {@code smallrye.faulttolerance.micrometer.disabled}
+     * MP Config property.
+     *
+     * @return A ShrinkWrap {@link Archive> containing a deployment that enables MP Telemetry
+     */
+    @Deployment(name = FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY_DISABLING_MICROMETER, managed = false)
+    public static Archive<?> deploymentWithMPTelemetryEnabledDisablingMicrometer() {
+        String mpConfig = "otel.service.name=" + FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY_DISABLING_MICROMETER
+                + "\notel.sdk.disabled=false\nTimeout/enabled=true\nsmallrye.faulttolerance.micrometer.disabled=true";
+        return ShrinkWrap.create(WebArchive.class, FAULT_TOLERANCE_DEPOYMENT_WITH_MP_TELEMETRY_DISABLING_MICROMETER + ".war")
+                .addPackages(true, HelloService.class.getPackage())
+                .addAsManifestResource(ConfigurationUtil.BEANS_XML_FILE_LOCATION, "beans.xml")
+                .addAsManifestResource(new StringAsset(mpConfig), "microprofile-config.properties");
+    }
+
+    /**
+     * Deploy all deployments, so it's possible to get their URLs in other tests. This is limitation of Arquillian
```

Comment:
Do we need to inject URL by arquillian? Host name and port are hard-coded already in many places around the TS and URL is http://127.0.0.1:8080/deployment-name. This workaround seems to be expensive ...

Reply:
We do need it. We can't allow for multiple deployments to be there at the same time, otherwise we would have un-deterministic behavior with respect to metrics collection.

The deployAll() method is something that we have to do if we want to deploy manually (required here) and be able to inject the URLs in the tests methods as Arquillian resources,

Reply:
Let's approve it now, I will double-check this next week and maybe create some follow-up issue ...